backend-rails/app/models/service_order.rb

```diff
@@ -29,4 +29,31 @@ class ServiceOrder < ApplicationRecord
   has_many :items, through: :item_service_orders, dependent: :destroy
 
   validates :status, inclusion: STATUSES
+
+  def products
+    products = []
+
+    item_service_orders.each do |item_service_order|
+      products << item_service_order if item_service_order.item.item_type == 'product'
+    end
+    products
+  end
+
+  def services
+    services = []
+
+    item_service_orders.each do |item_service_order|
+      services << item_service_order if item_service_order.item.item_type == 'service'
+    end
+    services
+  end
+
+  def total_price_items(items)
+    total = 0
+
+    items.each do |item_service_order|
```

Comment:
I think you can use sum instead of each

Reply:
hmmm, nice. I didn't remeber of sum method